### PR TITLE
Add new security events created by authentication-service

### PR DESCRIPTION
**Pull request by Soulou**
Related to [UFSSP-31] [UFSSP-30] [UFSSP-37]

[UFSSP-31]: https://scalingo.atlassian.net/browse/UFSSP-31?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UFSSP-30]: https://scalingo.atlassian.net/browse/UFSSP-30?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UFSSP-37]: https://scalingo.atlassian.net/browse/UFSSP-37?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: events_structs.go [hunk 1]
```diff
@@ -755,6 +803,71 @@ func (ev *EventDeleteNotifierType) String() string {
 	return fmt.Sprintf("Notifier '%s' deleted on %s app", d.NotifierName, ev.AppName)
 }
 
+// Security events
```
Comment (EtienneM) [inline events_structs.go:806]: We started splitting to multiple files the events. Maybe add all these security related events in a file `events_security.go`?

File: events_structs.go [hunk 2]
```diff
@@ -967,6 +1080,18 @@ func (pev *Event) Specialize() DetailedEvent {
 		e = &EventLinkGithubType{Event: ev}
 	case EventUnlinkGithub:
 		e = &EventUnlinkGithubType{Event: ev}
+	case EventLoginSuccess:
```
Comment (EtienneM) [inline events_structs.go:1083]: Shouldn't you add the DB related events in this switch / case (e.g. EventDatabaseAddFeature)?

File: events_boilerplate.go [hunk 3]
```diff
@@ -69,6 +69,12 @@ func (e *EventNewAddonType) TypeDataPtr() interface{} {
 func (e *EventUpgradeAddonType) TypeDataPtr() interface{} {
 	return &e.TypeData
 }
+func (e *EventDatabaseAddFeatureType) TypeDataPtr() interface{} {
```
Comment (EtienneM) [inline events_boilerplate.go:72]: Shouldn't you add to this file the login related events?

**Comment by Soulou** [comment]
I didn't create a `event_security.go` since the notion of security event will disappear, the RemoteIP will become general

File: events_structs.go (hunk 2 above)
Comment (EtienneM) [reply events_structs.go:912]: You should "Resolve" the conversations. The person who opened it is responsible for considering it closed.

In this case I don't see where you fixed this comment? This switch / case statement must contain the `EventDatabase*Feature` cases.

File: events_boilerplate.go (hunk 3 above)
Comment (EtienneM) [reply events_boilerplate.go:72]: You should "Resolve" the conversations. The person who opened it is responsible for considering it closed.

You didn't implement `TypeDataPtr` functions for the events in the `events_account.go` file.

**Review by EtienneM — CHANGES_REQUESTED** [review]

**Comment by Soulou** [comment]
Sorry for the "Resolve"..

File: events_boilerplate.go (hunk 3 above)
Comment (Soulou) [reply events_boilerplate.go]: Done through code generation

File: events_structs.go [hunk 4]
```diff
@@ -760,150 +770,3 @@ type EventTfaDisabledType struct {
...
-	case EventRegenerateToken:
-		e = &EventRegenerateTokenType{Event: ev}
-	case EventDeleteToken:
-		e = &EventDeleteTokenType{Event: ev}
-	case EventTfaEnabled:
-		e = &EventTfaEnabledType{Event: ev}
-	case EventTfaDisabled:
-		e = &EventTfaDisabledType{Event: ev}
-	// Deprecated events. Replaced by equivalent with SCM in the name instead of
-	// Github
```
Comment (EtienneM) [inline events_structs.go:895]: With the Go generator, with lose this information that these are deprecated events. Maybe move this comment near the `EventLinkGithub` and `EventUnlinkGithub` structure definition?

File: README.md [hunk 5]
```diff
@@ -53,11 +53,14 @@ instance if your event type is named `my_event`:
     * Add `EventMyEventType` structure which embeds a field `TypeData` of the
         type `EventMyEventTypeData`.
     * Implement function `String` for `EventMyEventType`
-    * Add support for this event type in the `Specialize` function
     * [optional] Implement `Who` function for `EventMyEventType`. E.g. if the
         event type can be created by an addon.
-* `events_boilerplate.go`: implement the `TypeDataPtr` function for the new
-    `EventMyEventType` structure.
+
+Once the Event hass been added run the following command to update boilerplate code
```
Comment (EtienneM) [inline README.md:59]: typo:
```suggestion
Once the Event has been added run the following command to update boilerplate code
```

File: events.go [hunk 6]
```diff
@@ -1,5 +1,8 @@
 package scalingo
 
+//go:generate go run cmd/gen_events_boilerplate/main.go
+//go:generate go run cmd/gen_events_specialize/main.go
```
Comment (EtienneM) [inline events.go:4]: I don't understand the purpose of this command. I assume it's a Go instruction thing but I don't know what they are.

File: events_account.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+type EventPasswordResetSuccessType struct {
+	Event
+	TypeData EventPasswordResetSuccessTypeData `json:"type_data"`
+}
+type EventPasswordResetSuccessTypeData EventSecurityTypeData
+
+func (ev *EventPasswordResetSuccessType) String() string {
+	return fmt.Sprintf("Password changed from from %v", ev.TypeData.RemoteIP)
```
Comment (EtienneM) [inline events_account.go:62]:
```suggestion
	return fmt.Sprintf("Password changed from %v", ev.TypeData.RemoteIP)
```

**Review by EtienneM — COMMENTED** [review]
Nice 🤩 

Just a few comments, but non-blocking ones. I'm already approving it

**Review by EtienneM — APPROVED** [review]

File: events_structs.go (hunk 4 above)
Comment (Soulou) [reply events_structs.go:895]: It's already present:

```
	// EventLinkGithub and EventUnlinkGithub events are kept for
	// retro-compatibility. They are replaced by SCM events.
	EventLinkGithub   EventTypeName = "link_github"
	EventUnlinkGithub EventTypeName = "unlink_github"
```

Comment (EtienneM) [reply events_structs.go:895]: Oh sorry, I searched for the string "deprecated" in the page 😅

File: events.go (hunk 6 above)
Comment (Soulou) [reply events.go:4]: Basically it says:

When `go generate` is run, run the following command, so just running some generators which were added to the project

**Pull request merged**